docs/source/usage.rst

```diff
@@ -0,0 +1,230 @@
+
+or with decorators:
+
+.. code-block:: python
+
+    from eav.decorators import register_eav
+
+    @register_eav
+    class Supplier(models.Model):
+        ...
+
+Generally, if you chose the former, the most appriopriate place for the
+statement would be at the bottom of your ``models.py`` or immmediately after
+model definition.
+
+Advanced Registration
+---------------------
+
+Under the hood, registration does a couple of things:
+
+1. Attaches :class:`~eav.managers.EntityManager` to your class. By default,
+   it replaces standard manager (*objects*). You can configure under which
+   attribute it is accessible with :class:`~.eav.registry.EavConfig` (see below).
+
+2. Binds your model's *post_init* signal with
+   :meth:`~eav.registry.Registry.attach_eav_attr` method. It is used to
+   attach :class:`~eav.models.Entity` helper object to each model instance.
+   Entity, in turn, is used to retrieve, store and validate attribute values.
+   By default, it's accessible under *eav* attribute:
+
+.. code-block:: python
+
+    part.eav.weight = 0.56
+    part.save()
+
+3. Binds your model's *pre_save* and *post_save* signals to
+   :meth:`~eav.models.Entity.pre_save_handler` and
+   :meth:`~eav.models.Entity.post_save_hander`, respectively.
+   Those methods are responsible for validation and storage
+   of attribute values.
+
+4. Setup's up generic relation to :class:`~eav.models.Value` set.
+   By default, it's accessed under *eav_values*:
+
+.. code-block:: python
+
+    patient.eav_values.all()
+    # = <QuerySet [has fever?: "True" (1), temperature: 37.7 (2)]>
+
+5. Sets *_eav_config_cls* attribute storing model of the config class
+   used by :class:`~eav.registry.Registry`. Defaults
+   to :class:`~eav.registry.EavConfig`; can be overridden (see below).
+
+With that out of the way, almost every aspect of the registration can
+be customized. All available options are provided to registration
+via config class: :class:`~eav.registry.EavConfig` passed to
+:meth:`~eav.register`. You can change them by overriding the class.
+Available options are as follows:
+
+1. ``manager_attr`` - Specifies manager name. Used to refer to the
+   manager from Entity class, "objects" by default.
+2. ``manager_only`` - Specifies whether signals and generic relation should
+   be setup for the registered model.
+3. ``eav_attr`` - Named of the Entity toolkit instance on the registered
+   model instance. "eav" by default. See attach_eav_attr.
+4. ``generic_relation_attr`` - Name of the GenericRelation to Value
+   objects. "eav_values" by default.
+5. ``generic_relation_related_name`` - Name of the related name for
+   GenericRelation from Entity to Value. None by default. Therefore,
+   if not overridden, it is not possible to query Values by Entities.
+
+Example registration may look like:
+
+.. code-block:: python
+
+    class SupplierEavConfig(EavConfig):
+        manager_attr = 'eav_objects'
+
+    eav.register(supplier, SupplierEavConfig)
```

Comment:
Is possible to use EavConfig classes with decorator registration as well? If so, there should be an example. If not, this should be mentioned, as the two registration possibilities are treated as equal at the top of this document.